upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js

```diff
@@ -115,6 +115,17 @@ function processEvent(event, callback) {
     
     var path = process.env.API_URL;
 
+    /**
+     * We only handle push events, of which there are many subtypes. Unfortunately, the only way to differentiate between them is to look
+     * for expected fields. There are no enums for push events subtypes.
+     * 
+     * If an event is deemed not supported, we will return a success and print a message saying the event is not supported.
+     */
+    if (!('repository' in body) || !('ref' in body) || !('created' in body) || !('deleted' in body) || !('pusher' in body)) {
```

Comment:
Double (quintuple) negation is confusing me. 
Might be easier to read transformed as 
`!(first condition and second condition and third condition ...) `
https://en.wikipedia.org/wiki/De_Morgan%27s_laws maybe

Second question: does this mean that if I add some repos and remove some repos at the same time `!(added and deleted)`, this will ignore that entirely?

Reply:
Does this do the same thing?
`if (['repository', 'ref', 'created', 'deleted', 'pusher'].some(str => !(str in body))) {`

Reply:
Each even only has one action, so you cannot add and remove some repos at the same time.

For example, adding a branch will have created = true and deleted = false, pushing a branch will have created = false, and deleted = false, and deleting a branch will have created = false and deleted = true

Reply:
@garyluu very fancy, I think that would work.